src/Persistence/Array_.php

```diff
@@ -146,17 +149,15 @@ public function tryLoadAny(Model $m, string $table = null): ?array
      *
      * @return mixed
      */
-    public function insert(Model $m, $data, string $table = null)
+    public function insert(Model $model, $data, string $table = null)
     {
-        if ($table === null) {
-            $table = $m->table;
-        }
+        $table = $table ?? $model->table;
 
-        $data = $this->typecastSaveRow($m, $data);
+        $data = $this->typecastSaveRow($model, $data);
 
-        $id = $this->generateNewId($m, $table);
-        if ($m->id_field) {
-            $data[$m->id_field] = $id;
+        $id = $this->generateNewId($model, $table);
+        if ($model->id_field) {
+            $data[$model->id_field] = $id;
```

Comment:
I know this is not directly related to Scopes, but if this is for AI, we should move it, and definitelly not assing to `$data`, `$data` must represent a map of table data by table name (and possibly, id_field can be a table name)

Reply:
Not sure what the point is. `$data` stores the complete data including rows